scripts/src/main/resources/scripts/functions

```diff
@@ -167,6 +167,102 @@ function doEchoInteraction() {
   echo -e "\033[96m${*}\033[39m"
 }
 
+# $1: version_start
+# $2: version_end
+# $3: version
+# returns 0 if version_start <= version <= version_end
+function doCheckVersionRange() {
+  local version1="${1}"
+  local version2="${2}"
+  local version="${3}"
+  doVersionCompare "${version}" "${version1}"
+  check1="${?}"
+  doVersionCompare "${version2}" "${version}"
+  check2="${?}"
+  if [ "${check1}" == "2" ] || [ "${check2}" == "2" ] 
+  then
+    return 1
+  else 
+    return 0
+  fi
+}
+
+# $1: software
+# $2: version
+# return 1 if version is found in tool's security file
+function doCheckSoftwareSecurityVersion() {
+  local software="${1}"
+  local version="${2}"
+  local security_file
+  local check="0"
+  local line=""
+  local myfirst
+  local mylast
+  if [ -d "${DEVON_IDE_HOME}/ide-urls" ]
+  then
+    if [ "${software}" == "git" ]
+    then 
+      edition="git"
+    else
+      if [ -z "${edition}" ]
+      then
+        edition="${software}"
+      fi
```

Comment:
I would recommend to provide the `edition` as optional 3rd parameter. If edition is empty, it will fallback to `software`:
```
local edition="${3}"
if [ -z "${edition}" ]; then
  edition="${software}"
fi
```

This makes the code more clear, less specific (no extra handling for `git`) and prevents side-effects via global variables.

Reply:
Done